src/rgw/driver/rados/rgw_sal_rados.cc

```diff
@@ -3835,9 +3846,9 @@ int RadosRole::delete_obj(const DoutPrefixProvider *dpp, optional_yield y)
 
 extern "C" {
 
-void* newRadosStore(void)
+void* newRadosStore(boost::asio::io_context* io_context)
 {
-  rgw::sal::RadosStore* store = new rgw::sal::RadosStore();
+  rgw::sal::RadosStore* store = new rgw::sal::RadosStore(*io_context);
```

Comment:
is `io_context` meant to be optional? can we take it by reference instead of pointer here?